src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,271 @@
+
+import duke.exception.DukeFileSystemException;
+import duke.exception.InvalidDukeCommandException;
+import duke.io.DukeParser;
+import duke.io.Storage;
+import duke.io.Ui;
+import duke.task.*;
+import duke.type.DukeCommand;
+
+
+/**
+ * A class that provides functionalities for users to run a command-line-interface task tracker.
+ */
+public class Duke {
+    // Constant declarations
+    private final Ui ui;
+    private final DukeParser parser;
+    private Optional<Storage> storage;
+    private TaskList taskList;
+
+
+    public Duke() {
+        String filePath = "data/duke.txt";
+        this.ui = new Ui();
+        this.parser = new DukeParser();
+        try {
+            this.storage = Optional.of(new Storage(filePath));
+            this.taskList = this.storage.map(value -> new TaskList(value.load())).orElseGet(TaskList::new);
+        } catch (IOException e) {
+            ui.dukeShowError("The file path " + filePath + " is invalid. This session will not be stored.");
+            this.storage = Optional.empty();
+            this.taskList = new TaskList();
+        } catch (DukeFileSystemException e) {
+            ui.dukePrint(e.getMessage());
+            this.taskList = new TaskList();
+        }
+    }
+
+    /**
+     * Constructs a new instance of Duke, the task manager. If the file path specified, which is used to store and load
+     * previously added tasks, is invalid, Duke will default to a new empty task list instead, and executing Duke.run
+     * () will not cache the tasks added or deleted for that particular session.
+     *
+     * @param filePath the file path to store and load previously added tasks
+     */
+    public Duke(String filePath) {
+        this.ui = new Ui();
+        this.parser = new DukeParser();
+        try {
+            this.storage = Optional.of(new Storage(filePath));
+            this.taskList = this.storage.map(value -> new TaskList(value.load())).orElseGet(TaskList::new);
+        } catch (IOException e) {
+            ui.dukeShowError("The file path " + filePath + " is invalid. This session will not be stored.");
+            this.storage = Optional.empty();
+            this.taskList = new TaskList();
+        } catch (DukeFileSystemException e) {
+            ui.dukePrint(e.getMessage());
+            this.taskList = new TaskList();
+        }
+    }
+
+    private String todoHandler(String args) {
+        if (args.equals("")) {
+            return "The description of a todo cannot be empty.";
+        }
+        taskList.appendTask(new Todo(args));
+        return String.format("Got it. I've added this task:\n%s\nNow you have %d task%s in the list.",
+                args, taskList.size(), taskList.size() > 1 ? "s" : "");
+    }
+
+    private String eventHandler(String args) {
+        if (args.equals("")) {
+            return "The description of an event cannot be empty.";
+        }
+        String taskDescription = args.split(" /at ")[0];
+        String eventTime;
```

Comment:
Since this eventTime variable is only used within the try block, it is better to declare it within the try block itself. Check out the [textbook](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#minimize-scope-of-variables) on minimizing scope of variables.

Reply:
Thanks for the comments!